app/assets/javascripts/application.js

```diff
@@ -14,6 +14,7 @@
 //= require angular-bootstrap/ui-bootstrap-tpls
 //= require angular-sanitize
 //= require angular.validators
+//= require ng-redux/dist/ng-redux
```

Comment:
Well this is because the main angular app is bootstrapped here [miq_angular_application.js#L1](https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/assets/javascripts/miq_angular_application.js#L1) and sadly you cannot add dependencies (modules), once app is loaded.

Reply:
We need to run `ng-redux` [main script](https://github.com/angular-redux/ng-redux/blob/master/src/index.js) which registers the `ngRedux` module and hook it to the Angular application module.

In `miq_angular_application.js` we declare `ngRedux` module as a dependency of `ManageIQ` (application) module. The important bit is that `miq_angular_application.js` is Rails-managed.

My understanding is that when Rails-managed `miq_angular_application.js` executes, the `ngRedux` module registration must be already done - and this implies that we need to load `ng-redux` library through `application.js ` file.

> you cannot add dependencies (modules), once app is loaded.

Officially yes, unofficially [there's a way](https://stackoverflow.com/a/37244717) but [it only works if you add those extra dependencies prior to application bootstrap](https://stackoverflow.com/a/32362944) (there are some undocumented APIs used by Angular injector).